retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
> "Total allocated" taken from memstats? 

From pproff.

> The "total" you get from pprof is the total amount of currently allocated heap. 

I was passing in `-alloc_space`

> if we can avoid them as easily as here.

I'm still not seeing how this would be easy.